### ISSUE TITLE
JPG filenames

### ISSUE THREAD

**Issue by profucius** [tallbl0nde/TMetaDump#1]
I tried the tool and it works great! This is so amazing for the project I want to do. Thank you for making it!

Just a tiny request if I may, is it possible to have the filenames be named the Title Name of the game? I had planned to use the icons in my front-end game manager, and I'm having some trouble thinking of a way to batch rename all of them from the TitleID to the game name on my PC after-the-fact.

Other than that, perfect tool. Thanks again

**Comment by tallbl0nde** [comment]
The issue with using the Title Name is that some of them have UTF-8 characters, or even just accented characters which the Switch FS module doesn't seem to support. This causes issues such as not being able to save the file, corruption, etc.

The JSON contains the relative path of each title's icon, so as long as you can parse the JSON it should be easy-ish to rename them automatically.

**Comment by profucius** [comment]
I understand, yeah I tried making a batch process today and ran into that too. 

I ended up coming up with a multi step process to batch this to fulfill my needs. If anyone else reads this and is curious how I did it, I basically just used Notepad++ to create a macro to parse the JSON file using RegEx to boil it down to a text list of game name and the title ID tabulated on the same line. Then I pasted that into Excel, title ID being in its own column, and then created a few columns for parsing using substitute and concatenate, the end result being a column with cells of windows commands for renaming a file from>to. "Ren filename1 filename2". Pasted the whole column into a .bat file and ran it. Almost all images filenames were converted correctly.

Thanks, I'll close this issue now.
